File: adv05.py
#!/usr/bin/env python3

with open("input5") as inp:
    instrs = list(map(int, inp.readlines()))

#instrs = [0, 3, 0, 1, -3]

count = 0
pos = 0

while pos >= 0 and pos < len(instrs):
    npos = pos + instrs[pos]
    if instrs[pos] >= 3:
        instrs[pos] -= 1
    else:
        instrs[pos] += 1
    count += 1
    pos = npos

print(count)

File: 2017/adv05.c
#define LEN 1066
int instrs[LEN];

main() {
  int pos;
  for (pos = 0; pos < LEN; pos++)
    scanf("%d", instrs+pos);

  int count = 0, npos;
  for (pos = 0; pos < LEN && pos >= 0; pos = npos) {
    int *ins = instrs + pos;
    npos = pos + *ins;
    if (*ins >= 3)
      (*ins)--;
    else
      (*ins)++;
    count++;
  }
  printf("%d\n", count);
}
